prompb/types.proto

```diff
@@ -56,6 +56,9 @@ message ReadHints {
   string func = 2;    // String representation of surrounding function or aggregation.
   int64 start_ms = 3; // Start time in milliseconds.
   int64 end_ms = 4;   // End time in milliseconds.
+  repeated string grouping = 5; // List of label names of any fields used in aggregation.
+  bool by = 6; // Indicate whether it is without or by, it will be false if grouping is not used.
+  string ta = 7; // Temporal aggregation read hint include func name and time window, such as rate:5m0s.
```

Comment:
No need for the function name, we just need the time and it should be in milliseconds like the other. The field name also doesn't go over the wire, so no need to abbreviate it. `range` is what we'd usually call this.

Reply:
@brian-brazil If we don't include function name, how can we distinguish `sum by (dim) (rate(range_vector))` and `avg by (dim) (rate(range_vector))`? `sum by (dim) (sum_over_time(range_vector))` and `max by (dim) (sum_over_time(range_vector))`?

Reply:
The function name is already in `func`

Reply:
@brian-brazil In this PR, we'd like to support 1) temporal aggregation, 2) spatial aggregation 3) temporal aggregation. For 1) & 2), the `func` will be temporal aggregation func name (such as `rate`, `sum_over_time`, etc.) name or spatial aggregation func name (such as `sum`, `avg`, etc.). However, for 3), the `func` will be spatial aggregation func name, we don't have a place to record the temporal aggregation func name.
For example:
`sum by (dim) (rate(range_vector))`, if we use `rate` as `func`, then we cannot identify spatial aggregation name `sum`.

Reply:
We could make func a list, but we're already quite bit of scope creep here so I'd say let's get this in without worrying about 3.

Reply:
make sense, let me make the change